kubernetes-discovery-client/src/test/groovy/io/micronaut/kubernetes/discovery/KubernetesDiscoveryClientFilterSpec.groovy

```diff
@@ -28,8 +31,8 @@ class KubernetesDiscoveryClientFilterSpec extends KubernetesSpecification implem
         serviceIds.contains("example-client")
 
         and:
-        Flowable.fromPublisher(discoveryClient.getInstances("example-client")).count().blockingGet() == 1
-        Flowable.fromPublisher(discoveryClient.getInstances("example-service")).count().blockingGet() == 0
+        Flowable.fromPublisher(discoveryClient.getInstances("example-client")).blockingFirst().size() == 1
```

Comment:
after the `defaultIfEmpty` there's always a list returned from the getInstances. So the tests needed to be fixed to reflect this and now the number of returned service instances is tested.